### PR TITLE
FIX-#3306: Move Modin XGBoost docs under `Experimental Features`

### LINKED ISSUE #3306: Move "Distributed XGBoost on Modin (experimental)" topic under "EXPERIMENTAL FEATURES" section.
Modin version: 54898ef78e83209fba3233606841ca988e531aee

See https://modin.readthedocs.io/en/latest/modin_xgboost.html.

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3306  
- [ ] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/experimental_features/index.rst [hunk 1]
```diff
@@ -39,6 +39,13 @@ SQL on Modin Dataframes
 
 Read about Modin Dataframe support for SQL queries in this recent `blog post`_. Check out the `Modin SQL documentation`_ as well!
 
+Distributed XGBoost on Modin
+----------------------------
+Modin provides an implementation of distributed XGBoost machine learning algorithm on Modin DataFrames. See our
+:doc:`Distributed XGBoost on Modin documentation </experimental_features/modin_xgboost>` for details about installation and usage and
```
Comment (YarShev) [inline docs/experimental_features/index.rst:45]:
```suggestion
:doc:`Distributed XGBoost on Modin documentation </experimental_features/modin_xgboost>` for details about installation and usage, as well as
```

**Review by YarShev — COMMENTED** [review]
Left a comment, the rest looks good!

File: docs/experimental_features/index.rst (hunk 1 above)
Comment (prutskov) [reply docs/experimental_features/index.rst]: fixed

**Review by YarShev — APPROVED** [review]
@prutskov , LGTM, thanks!

**Pull request merged**